app/views/notifier/invite.markerb

```diff
@@ -1 +1,8 @@
-<%= t('.message', :invite_url => invite_code_url(@invitation_code), :diasporafoundation_url => 'https://diasporafoundation.org/', :diaspora_id => @inviter.try(:diaspora_handle)) %>
+<%= t('.message',
+  invite_url: invite_code_url(@invitation_code),
+  diasporafoundation_url: 'https://diasporafoundation.org/',
+  user: @inviter.try(:diaspora_handle) == @inviter.try(:name) ?
+    @inviter.try(:diaspora_handle) :
+    @inviter.try(:name) + " (" + @inviter.try(:diaspora_handle) + ")",
```

Comment:
How about combining your proposal with a method #name_with_handle or #pretty_name or #humanize_handle that returns:

~~~
# Return the user's named (if set) and their handle 
def name_with_handle
  if self.name.present?
    return "#{self.name} (#{self.handle})"     # User (userhandle@domain.tld)
  else
    return "#{self.handle}"                    # userhandle@domain.tld
  end
end
~~~